### PR TITLE
Support model comparing

### PR DESCRIPTION
**Pull request by pminev**
The model comparing goes through 3 phases:
1. Logit comparison of vector length - we check if the logit vector's are very similar to their length
2. Token comparison - we check that at least 80% of the tokens are the same. We're not checking for 100% because it might happen if 2 tokens have very similar logits will be misplaced due to the floating precision error(might happen on different machines).
3. Checking the divergence of each token's probability by using [Jensen-Shannon algorithm](https://en.wikipedia.org/wiki/Jensen–Shannon_divergence).

I was able to hit an edge case for similar models(llama3 8b vs llama 70b)where the probability was 1, and their logits were almost the same, but in all other runs their results were different and the metric was OK.

Linked with: https://github.com/alpaca-core/acord/pull/21

### CODE REVIEW

File: code/ac/llama/LogitComparer.hpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Alpaca Core
+// SPDX-License-Identifier: MIT
+//
+#pragma once
+#include "Token.hpp"
+#include <unordered_map>
+
+namespace ac::llama {
+
+class LogitComparer {
```
Comment (pminev) [inline code/ac/llama/LogitComparer.hpp:10]: Maybe we should rename this class since it's not only legit comparison only? Does `TokenDataComparer` sounds better?

File: code/ac/llama/Token.hpp [hunk 2]
```diff
@@ -3,8 +3,17 @@
 //
 #pragma once
 #include <cstdint>
+#include <vector>
 
 namespace ac::llama {
 using Token = std::int32_t;
 inline constexpr Token Token_Invalid = -1;
+
+struct TokenData {
```
Comment (pminev) [inline code/ac/llama/Token.hpp:12]: I've added this here since I needed it in the `Sampler` and `Session`. I'm not sure if here is the best place to stay, so I'm open for ideas how to move it.

**Review by iboB — APPROVED** [review]
Don't use acord for io. You can set this up much more easily with
